### PR TITLE
CLI command to retrieve syncing status ethereum blockhain

### PR DESCRIPTION
**Pull request by DylanVerstraete**
reference: https://github.com/threefoldfoundation/tfchain/issues/303

Needs fix:
- Lets user execute command `tfchain erc20 syncstatus` when the deamon is starting up but not synced yet (currently not possible, see https://github.com/threefoldfoundation/tfchain/issues/297)

### CODE REVIEW

File: cmd/tfchainc/erc20cmd.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			Long:  `Get the ethereum chain sync status.`,
+			Run:   erc20SubCmds.getSyncingStatus,
+		}
+	)
+
+	rootCmd.AddCommand(getSyncingStatusCmd)
+
+	// register flags
+	getSyncingStatusCmd.Flags().Var(
+		cli.NewEncodingTypeFlag(0, &erc20SubCmds.getSyncingStatusCfg.EncodingType, 0), "encoding",
```
Comment (GlenDC) [inline cmd/tfchainc/erc20cmd.go]: `0` allows every kind of encoding type, we only want to allow JSON and Human-readable though.

File: cmd/tfchainc/erc20cmd.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	syncingStatus, err := erc20cmds.GetSyncingStatus()
+	if err != nil {
+		cli.DieWithError("error while fetching the syncing status", err)
+	}
+
+	// encode depending on the encoding flag
+	var encode func(interface{}) error
+	switch erc20SubCmds.getSyncingStatusCfg.EncodingType {
+	case cli.EncodingTypeHuman:
```
Comment (GlenDC) [inline cmd/tfchainc/erc20cmd.go]: You go way too complex here. I would do the encoding something as follows:

```go
switch erc20SubCmds.getSyncingStatusCfg.EncodingType {
case cli.EncodingTypeHuman:
     if not-syncing {
          fmt.Println("ERC20 node is not synchronizing") // TODO: we might want to still be able to print current height somehow
     } else {
          fmt.Printf(`ERC20 node is currently synchronizing...
Starting Height: %d
Current Height: %d,
blabla: %d
`, val, val, val)
      }
case cli.EncodingTypeJSON:
    err := json.NewEncoder(os.Stdout).Encode(syncingStatus)
    if err != nil {
          cli.DieWithError("failed to encode syncing status", err)
     }
}
```

File: cmd/tfchainc/erc20cmd.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+type erc20SubCmds struct {
+	cli                 *internal.CommandLineClient
+	getSyncingStatusCfg struct {
+		EncodingType cli.EncodingType
+	}
+}
+
+func (erc20SubCmds *erc20SubCmds) getSyncingStatus(cmd *cobra.Command, args []string) {
+	erc20cmds := internal.NewERC20Client(erc20SubCmds.cli)
```
Comment (GlenDC) [inline cmd/tfchainc/erc20cmd.go]: There is honestly no need to make an entire client here. I would simply call the endpoint directly here. You're adding an abstraction layer here without any benefit.

File: cmd/tfchainc/erc20cmd.go (hunk 2 above)
Comment (DylanVerstraete) [reply cmd/tfchainc/erc20cmd.go]: Changed in: https://github.com/threefoldfoundation/tfchain/pull/307/commits/dc958a46503e7ab519c9d29c5d94598f09ea7d5c

File: cmd/tfchainc/erc20cmd.go (hunk 1 above)
Comment (DylanVerstraete) [reply cmd/tfchainc/erc20cmd.go]: Changed in: https://github.com/threefoldfoundation/tfchain/pull/307/commits/5e3a7be7c727594b63955708990632c0e1ddce8f

File: cmd/tfchainc/erc20cmd.go (hunk 3 above)
Comment (DylanVerstraete) [reply cmd/tfchainc/erc20cmd.go]: Changed in: https://github.com/threefoldfoundation/tfchain/pull/307/commits/bdce1a1b3d82ccecd576e9aa8f99c92d3f270699

**Pull request merged**